src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/ISyntaxFactsService.cs

```diff
@@ -286,10 +287,14 @@ internal interface ISyntaxFactsService : ILanguageService
         bool IsTopLevelNodeWithMembers(SyntaxNode node);
         bool HasIncompleteParentMember(SyntaxNode node);
 
+        bool IsPureBlock(SyntaxNode node);
         bool IsExecutableBlock(SyntaxNode node);
         SyntaxList<SyntaxNode> GetExecutableBlockStatements(SyntaxNode node);
         SyntaxNode FindInnermostCommonExecutableBlock(IEnumerable<SyntaxNode> nodes);
 
+        bool IsStatementContainer(SyntaxNode node);
+        IReadOnlyList<SyntaxNode> GetStatementContainerStatements(SyntaxNode node);
```

Comment:
Wouldn't this always return a SyntaxList?

Reply:
No, not for a C# embedded statement. It will be wrapped in an immutable array.